app/models/manageiq/providers/ibm_cloud/vpc/cloud_manager/vm.rb

```diff
@@ -112,15 +112,69 @@ def raw_destroy
     unsupported_reason_add(:resize, _('The VM is not powered off')) unless current_state == "off"
   end
 
-  def raw_resize(new_flavor)
+  def raw_resize(options)
     with_provider_object do |instance|
-      instance.resize(instance, new_flavor.name)
+      instance.resize(instance, options["flavor"])
     rescue IBMCloudSdkCore::ApiException => e
       Notification.create(:type => :vm_resize_error, :options => {:subject => instance[:name], :error => e.error})
       raise MiqException::MiqProvisionError, e.to_s
     end
   end
 
+  def params_for_resize
+    {
+      :fields => [
+        {
+          :component  => 'select',
+          :name       => 'flavor',
+          :id         => 'flavor',
+          :label      => _('Flavor'),
+          :isRequired => true,
+          :options    => resize_form_options.map do |flavor|
+            {
+              :label => flavor[:name],
+              :value => flavor[:id],
+            }
+          end,
+        },
+      ],
+    }
+  end
+
+  def resize_form_options
+    flavors = []
+    # include only flavors with root disks at least as big as the instance's current root disk.
+    self.ext_management_system&.flavors&.each do |ems_flavor|  
+      # include only flavors with root disks at least as big as the instance's current root disk.
+      if self.flavor.nil? || ((ems_flavor != self.flavor) && (ems_flavor.root_disk_size >= self.flavor.root_disk_size))
+        flavors << {:name => ems_flavor.name_with_details, :id => ems_flavor.name}
+      end
+    end
+    flavors
```

Comment:
Few things:
1. In ruby it is preferred to use:
```
array.map do |item|
  {:key => item.value}
end
```

Instead of:
```
result = []
array.each do |item|
  result << {:key => item.value}
end
```

2. Since we're building an array of hashes we might as well return it in the format that is expected by `:options => ` above

3. If you use `.map` then the implicit return will work without having to explicitly return `flavors`

4. The `self.ext_management_system` is redundant (surprised miq-bot hasn't commented yet about this)

5. It looks like the logic might be a little cleaner if we invert the conditional below
```suggestion
    # include only flavors with root disks at least as big as the instance's current root disk.
    ext_management_system.flavors.map do |ems_flavor|
      # include only flavors with root disks at least as big as the instance's current root disk.
      next if flavor && (ems_flavor == flavor || ems_flavor.root_disk_size < flavor.root_disk_size)

      {:label => ems_flavor.name_with_details, :id => ems_flavor.name}
    end
```